src/main/java/org/folio/rest/impl/AuthorityNoteTypeAPI.java

```diff
@@ -0,0 +1,57 @@
+package org.folio.rest.impl;
+
+import java.util.Map;
+import javax.ws.rs.core.Response;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Handler;
+
+import org.folio.rest.annotations.Validate;
+import org.folio.rest.jaxrs.model.AuthorityNoteType;
+import org.folio.rest.jaxrs.model.AuthorityNoteTypes;
+import org.folio.rest.persist.PgUtil;
+
+public class AuthorityNoteTypeAPI implements org.folio.rest.jaxrs.resource.AuthorityNoteTypes {
+  public static final String REFERENCE_TABLE  = "authority_note_type";
+
+  @Override
+  @Validate
+  public void getAuthorityNoteTypes(String query, int offset, int limit, String lang,
+      Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    PgUtil.get(REFERENCE_TABLE, AuthorityNoteType.class, AuthorityNoteTypes.class, query, offset, limit, okapiHeaders,
+      vertxContext, GetAuthorityNoteTypesResponse.class, asyncResultHandler);
+  }
+
+  @Override
+  @Validate
+  public void postAuthorityNoteTypes(String lang, AuthorityNoteType entity, Map<String, String> okapiHeaders,
+      Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    PgUtil.post(REFERENCE_TABLE, entity, okapiHeaders, vertxContext, PostAuthorityNoteTypesResponse.class,
+      asyncResultHandler);
+  }
+
+  @Override
+  @Validate
+  public void getAuthorityNoteTypesById(String id, String lang, Map<String, String> okapiHeaders,
+      Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    PgUtil.getById(REFERENCE_TABLE, AuthorityNoteType.class, id,
+      okapiHeaders, vertxContext, GetAuthorityNoteTypesByIdResponse.class, asyncResultHandler);
+  }
+
+  @Override
+  @Validate
+  public void deleteAuthorityNoteTypesById(String id, String lang, Map<String, String> okapiHeaders,
```

Comment:
Do we need to restrict deletion default note types?

Reply:
Khalila said is not necessarily